src/converter/one/jfr/event/MallocLeakAggregator.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package one.jfr.event;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Map;
+import java.util.HashMap;
+
+public class MallocLeakAggregator implements IEventAggregator {
+    private final IEventAggregator wrapped;
+    private List<Event> events;
+    private double grain = 0;
+
+    Map<Long, MallocEvent> addresses = new HashMap<>();
+
+    public MallocLeakAggregator(IEventAggregator wrapped) {
+        this.wrapped = wrapped;
+        this.events = new ArrayList<Event>();
+    }
+
+    private List<Event> filter(List<Event> events) {
+        for (Event event : events) {
+            if (!(event instanceof MallocEvent)) {
+                continue;
+            }
+
+            MallocEvent e = (MallocEvent) event;
+            if (e.size > 0) {
+                addresses.put(e.address, e);
+            } else {
+                addresses.remove(e.address);
+            }
+        }
+
+        return new ArrayList<>(addresses.values());
```

Comment:
The list of "filtered" events is not actually used, is it? Therefore, creating `ArrayList` seems redundant.

Reply:
It is used in multi-chunk jfrs, filtered events will be sorted together with new events, before filtering again.